<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MODBUS_client</class>
 <widget class="QWidget" name="MODBUS_client">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>371</width>
    <height>409</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QFrame" name="frame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>client</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>client</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cb_port"/>
           </item>
           <item>
            <widget class="QToolButton" name="btn_refresh">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QPushButton" name="btn_connect">
             <property name="text">
              <string>connect</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="btn_disconnect">
             <property name="text">
              <string>disconnect</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>test</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_6">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>StartAddress:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_start_address"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="4" column="2">
            <widget class="QPushButton" name="btn_test_write_holding_registers">
             <property name="text">
              <string>write</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="btn_test_read_coils">
             <property name="text">
              <string>read</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="btn_test_write_coils">
             <property name="text">
              <string>write</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="btn_test_read_discrete_inputs">
             <property name="text">
              <string>read</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="btn_test_write_discrete_inputs">
             <property name="text">
              <string>write</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QPushButton" name="btn_test_read_holding_registers">
             <property name="text">
              <string>read</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QPushButton" name="btn_test_read_input_registers">
             <property name="text">
              <string>read</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QPushButton" name="btn_test_write_input_registers">
             <property name="text">
              <string>write</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Discrete Inputs</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Coils</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Input Registers</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Holding Registers</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>cb_port</tabstop>
  <tabstop>btn_refresh</tabstop>
  <tabstop>btn_connect</tabstop>
  <tabstop>btn_disconnect</tabstop>
  <tabstop>sb_start_address</tabstop>
  <tabstop>btn_test_read_discrete_inputs</tabstop>
  <tabstop>btn_test_write_discrete_inputs</tabstop>
  <tabstop>btn_test_read_coils</tabstop>
  <tabstop>btn_test_write_coils</tabstop>
  <tabstop>btn_test_read_input_registers</tabstop>
  <tabstop>btn_test_write_input_registers</tabstop>
  <tabstop>btn_test_read_holding_registers</tabstop>
  <tabstop>btn_test_write_holding_registers</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
